hubspot3/tickets.py

```diff
@@ -78,3 +90,171 @@ def get_all(self, limit: int = -1, **options) -> list:
+                yield changes
+
+    def _parse_changes(self, batch, query_limit, max_time_diff_ms):
+        # First lets group changes by deal id
+        changes_by_id = [(ticket_change['objectId'], ticket_change) for ticket_change in batch]
+        changes_by_id = sorted(changes_by_id, key=itemgetter(0))
+        changes_by_id = dict((key, list(value)) for key, value in
+                             itertools.groupby(changes_by_id, key=itemgetter(0)))
+        # Loop through ids in batches of 100 (limit of the batch API)
+        for index in range(int(len(changes_by_id) / query_limit) + 1):
+            # Get a batch of 100 elements
+            changes_by_id_batch = dict(list(changes_by_id.items())[index * query_limit:(index + 1)
+                                                                   * query_limit])
+            properties = []
+            for ticket_id, changes in changes_by_id_batch.items():
+                for ticket_id, change in changes:
+                    if change['changeType'] != TicketsClient.Recency.DELETED:
+                        properties += change['changes']['changedProperties']
+            properties = set(properties)
+            # Get the information of each change
+            tickets_history = self.get_batch_with_history(list(changes_by_id_batch.keys()),
+                                                          extra_properties=list(properties))
+            for ticket_id, ticket_history in tickets_history.items():
+                ticket_id = int(ticket_id)
+                properties = ticket_history['properties']
+                # Match the information of changes with each change
+                changes = changes_by_id_batch[ticket_id]
+                for ticket_id, change in changes:
+                    change['changes']['changedValues'] = {}
+                    for changed_variable in change['changes']['changedProperties']:
+                        versions = properties[changed_variable]['versions']
+                        time_diffs = [abs(version['timestamp'] - change['timestamp'])
+                                      for version in versions]
+                        min_time_index = min(range(len(time_diffs)),
+                                             key=time_diffs.__getitem__)
+                        if time_diffs[min_time_index] <= max_time_diff_ms:
+                            value = versions[min_time_index]['value']
+                            change['changes']['changedValues'][changed_variable] = value
+                        else:
+                            get_log(__name__).warning(f"Unable to find value for {changed_variable}"
+                                                      f" within the time range for ticket"
+                                                      f" {ticket_id}")
+        # Finally lets return only the changes as a list
+        return [change[1] for changes in changes_by_id.values() for change in changes]
```

Comment:
I don't think this is the best way to flatten the output, but, if it is, you could explain the meaning of the number `1`.